src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/MemberExpansion.cs

```diff
@@ -27,7 +27,16 @@ internal sealed class MemberExpansion : Expansion
             Predicate<MemberInfo> predicate,
             Formatter formatter)
         {
-            var runtimeType = value.Type.GetLmrType();
+            // For members of type DynamicProperty (part of Dynamic View expansion), we want
+            // to expand the underlying value (not the members of the DynamicProperty type).
+            var type = value.Type;
+            var isDynamicProperty = type.GetLmrType().IsDynamicProperty();
+            if (isDynamicProperty)
+            {
+                value = value.GetFieldValue("value", inspectionContext);
+            }
```

Comment:
It does, but this is a special internal type that is only returned by the DMODebugView.  Unless you object, I'd rather not have to thread a flag through to tell us if this is part of a Dynamic View expansion.